doc/ug/sec_cm_dv_framework/index.md

```diff
@@ -0,0 +1,128 @@
+---
+title: "Security Countermeasure Verification Framework"
+---
+
+## Purpose
+In a security chip, there are many various hardening security countermeasures implemented in security blocks.
+Hardened security countermeasures are used to detect attacks by embedding integrity checks.
+Usually it can’t be triggered by software or toggling the pins on the chip.
+So, this may need to be tested one by one via forcing internal design signals on the security countermeasure related logic, which mimics attackers to flip the circuit.
+This kind of verification is needed but the effort could be tremendous.
+The work can be significantly reduced by using a unified verification framework.
+
+## Standardized Design Countermeasure Primitive
+Each countermeasure is implemented as a primitive - an RTL design building block written in SystemVerilog.
+The standardization of how these building blocks are designed paves the way for automated verification.
+
+There are 2 requirements that are needed in this primitive.
+1. Create a standard primitive for each kind of countermeasure, such as a security counter module, which can detect if the counter value is modified unintentionally.
+These primitives could be reused across all blocks in the chip.
+Each primitive has an error indicator as an IO (or an internal signal if other blocks don’t need to connect with this error indicator).
+Once the design detects an attack, the error will be set.
+
+![diagram_of_sec_cm_prim](diagram_of_sec_cm_prim.svg)
+
+2. Once the error is set, the block should report a fatal alert or trigger unmasked interrupt to the system.
+There could be some other logics which are triggered when error occurs but not unified across the blocks.
+![diagram_of_block_that_contains_sec_cm_primitive](diagram_of_block_that_contains_sec_cm_primitive.svg)
+
+## Embedded Common Checks In The Design
+At the top of each IP, an assertion check should be added to ensure that the fatal alert will be triggered once error is set.
+In the countermeasure primitive, there is another check to ensure that each instance of the countermeasure primitive always has this assertion check.
```

Comment:
"always has this assertion check."... seems unclear, you mean something like "is connected to the fatal alert"?

Reply:
I actually mean the instance of prim contains the error-trigger-alert assertion check.
Reworded a bit.